modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -810,6 +812,16 @@ protected void paintComponent(Graphics g) {
 
         Graphics gg = null;
         try {
+            ComponentOrientation cor = this.getComponentOrientation();
+            if (!cor.equals(ComponentOrientation.UNKNOWN)) {
+                Scene scene = getScene();
+                if (scene != null) {
+                    boolean rtl = cor.equals(ComponentOrientation.RIGHT_TO_LEFT);
+                    scene.setNodeOrientation(rtl ?
+                                             NodeOrientation.RIGHT_TO_LEFT :
+                                             NodeOrientation.LEFT_TO_RIGHT);
```

Comment:
I have tried to put in place what you suggested. Can you please take a look at the PR and see if it's in the right direction and suggest further.
I have few doubts

- It seems JFXPanel call EmbeddedStage interface methods and not the other way round, so I am not sure how to "call a method in embedded stage that would delegate to JFXPanel".
- It seems [Scene::calcEffectiveNodeOrientation](https://github.com/openjdk/jfx/blob/9b93c962f45e5cf0b3a9f1090f307603be130a0e/modules/javafx.graphics/src/main/java/javafx/scene/Scene.java#L6343) is private and not part of swing-interop code/package so I am not sure how to leverage it from JFXPanel/EmbeddedStage....I have tried setting EmbeddedStage orientation from JFXPanel and query it from Scene but am not sure how to call `calcEffectiveNodeOrientation`

Reply:
I have made the suggested changes and updated the PR.
But it seems the orientation of the text in the test attached in JBS is at the middle initially (and not at LTR or RTL) and it only begins to track the orientation after 1 iteration of changing JFrame orientation. Is it because of INHERIT default value which causes the effective orientation to not follow the changed orientation immediately? Does it need to be looked at or is it expected?

Reply:
This might be due to the initialization problem I pointed out or it might be for some other reason. Can you make the suggested change and then re-test?

Reply:
OK. Please file a new testbug to create a regression test for this.

Reply:
Yes, filed https://bugs.openjdk.org/browse/JDK-8319147